zrml/prediction-markets/src/lib.rs

```diff
@@ -139,13 +152,30 @@ mod pallet {
             let market_status = market.status;
             let outcome_assets = Self::outcome_assets(market_id, &market);
             let outcome_assets_amount = outcome_assets.len();
+
+            // Slash outstanding bonds
+            let slash_market_creator = |amount| {
+                CurrencyOf::<T>::slash_reserved_named(&RESERVE_ID, &market.creator, amount);
+            };
+            if market_status == MarketStatus::Proposed {
+                slash_market_creator(T::AdvisoryBond::get());
+            }
+            if market_status != MarketStatus::Resolved
+                && market_status != MarketStatus::InsufficientSubsidy
+            {
+                if market.creation == MarketCreation::Permissionless {
+                    slash_market_creator(T::ValidityBond::get());
+                }
+                slash_market_creator(T::OracleBond::get());
+            }
```

Comment:
```suggestion
            match market_status {
                // already unreserved or slashed when InsufficientSubsidy or Resolved
                MarketStatus::InsufficientSubsidy | MarketStatus::Resolved => (),
                // reserves left when market is proposed (when proposed always advised then)
                MarketStatus::Proposed if market.creation == MarketCreation::Advised => {
                    slash_market_creator(T::AdvisoryBond::get())
                }
                _ => {
                    // for permissionless and advised markets and
                    // when not InsufficientSubsidy or Resolved then the OracleBond is left
                    slash_market_creator(T::OracleBond::get());
                    // when not InsufficientSubsidy or Resolved for a permissionless market the ValidityBond is left 
                    if market.creation == MarketCreation::Permissionless {
                        slash_market_creator(T::ValidityBond::get());
                    }
                }
            }
```
I agree with your style. The match would be my style. Probably you should comment a little bit in there, because it's hard to understand (when looking at it in two months from now).

Reply:
Not a big fan of this comment style, but I've added a clarifying comment.